### DISCUSSION TITLE
_console and MPI rank!=0 output

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#16333] [Q&A General] [answered]
Hi everyone,

I have an object derived from `Output` (and `UserObjectInterface`, if that makes any difference), that writes stuff to `_console`.  Everything works as planned for the MPI rank=0 processor.  However, i *believe* the rank!=0 processors' `_console` stream does not get sent to std::out.   How do i instruct the rank!=0 `_console` to be written to std::out ?

Note - it's quite late at night (for me) so i could be completely wrong about this rank!=0 stuff - perhaps there's another bug in my code.  So don't spend too long thinking about this if it doesn't make sense to you!

a

**Comment by WilkAndy** [comment, marked as answer]
OK, i just found the answer:

```
my_app-opt -i input.i --keep-cout
```

**Reply by WilkAndy** [reply]
and my notes demonstrate that i must have asked this before, duh

**Reply by lindsayad** [reply]
I'm going to start asking a bunch of questions I know the answer to in order to climb the Most Answers leaderboard :smile: Just kidding, you nailed it!

**Reply by WilkAndy** [reply]
Just stay up way past your bedtime....
